### PR TITLE
feat(shell): add unknown_indicator parameter

### LINKED ISSUE #2541: Allow setting some default value for Shell prompt
#### Is your feature request related to a problem? Please describe.
Sometimes I use Starship with the Command Prompt on Windows or the `sh` shell on Linux. The Shell prompt doesn't display anything there.

#### Describe the solution you'd like
Simple solution: allow setting a `default` option in the config file.

#### Describe alternatives you've considered
_None_

#### Additional context
I see that the Shell prompt is also missing the `style` option. That can be added too!

### PR DESCRIPTION
**Pull request by rashil2000**
This adds a default parameter for the shell module, which is displayed in case Starship is not able to identify the shell.

#### Description
Fairly simple addition of a 'unknown-indicator' element in the ShellConfig struct. 
Please note that I'm a complete beginner in Rust, so if there's any mistakes, I'll be glad to rectify them 😅

#### Motivation and Context

Closes #2541 

#### How Has This Been Tested?
Running `cargo test` passes all existing tests. Let me know if I should add one more test case (like `test_default_if_unknown_shell`).
- [ ] I have tested using **MacOS**
- [ ] I have tested using **Linux**
- [x] I have tested using **Windows**

#### Checklist:

- [x] I have updated the documentation accordingly.
- [ ] I have updated the tests accordingly.

### CODE REVIEW

File: src/configs/shell.rs [hunk 1]
```diff
@@ -13,6 +13,7 @@ pub struct ShellConfig<'a> {
     pub ion_indicator: &'a str,
     pub elvish_indicator: &'a str,
     pub tcsh_indicator: &'a str,
+    pub default: &'a str,
```
Comment (andytom) [inline src/configs/shell.rs:16]: Personally I would prefer to use a more specific name for this option maybe `unknown_indicator`?

Comment (rashil2000) [reply src/configs/shell.rs:16]: Sure, that certainly makes more sense! I'll change it.

File: src/configs/shell.rs [hunk 2]
```diff
@@ -27,6 +28,7 @@ impl<'a> Default for ShellConfig<'a> {
             ion_indicator: "ion",
             elvish_indicator: "esh",
             tcsh_indicator: "tsh",
+            unknown_indicator: "",
```
Comment (vladimyr) [inline src/configs/shell.rs:31]: Does it make sense to use the question mark here as default? :thinking:

Comment (rashil2000) [reply src/configs/shell.rs:31]: Leaving it empty would be, in my opinion, slightly less jarring than a question mark.

Comment (vladimyr) [reply src/configs/shell.rs:31]: I guess you are right, especially because this is a new option.

**Review by vladimyr — APPROVED** [review]
LGTM

**Comment by rashil2000** [comment]
Not sure why the test is failing, I haven't touched the `src/configs/mod.rs` file. Do I need to do something?

**Comment by vladimyr** [comment]
Oh that's not your fault, can you please merge latest master?

**Review by andytom — APPROVED** [review]

**Comment by andytom** [comment]
Thank you for your contribution @rashil2000 and thank you reviewing @vladimyr

**Pull request merged**